src/EventGridExtension/OutputBinding/EventGridOutputAttribute.cs

```diff
@@ -0,0 +1,24 @@
+﻿using System;
+using Microsoft.Azure.WebJobs.Description;
+
+namespace Microsoft.Azure.WebJobs.Extensions.EventGrid
+{
+    /// <summary></summary>
+    /// <seealso cref="System.Attribute" />
+    [AttributeUsage(validOn: AttributeTargets.Parameter)]
+    [Binding]
+    public sealed class EventGridOutputAttribute : Attribute
+    {
+#pragma warning disable CS0618 // Type or member is obsolete
+        /// <summary>Gets or sets the topic hostname setting. Eg: topic1.westus2-1.eventgrid.azure.net</summary>
+        [AppSetting]
+#pragma warning restore CS0618 // Type or member is obsolete
+        public string TopicHostname { get; set; }
```

Comment:
looking at the doc, I think we should let user provide endpoint and extract the hostname in the code
https://github.com/Azure/azure-docs-sdk-dotnet/blob/master/api/overview/azure/eventgrid.md#sample-usage